tasks/main.yml

```
@@ -1,162 +1,65 @@
-
-- name: copy arch.make
-  template:
-    src: full_options.make
-    dest: "{{ siesta_topdir }}/Obj/arch.make"
-
-- name: Make siesta executables
-  shell: "make"
-  args:
-    chdir: "{{ siesta_topdir }}/Obj"
-    creates: "{{ siesta_topdir }}/Obj/siesta"
-  register: siesta_make
-
-- name: Make COOP/COHP utility
-  shell: "make mprop"
-  args:
-    chdir: "{{ siesta_topdir }}/Util/COOP"
-    creates: "{{ siesta_topdir }}/Util/COOP/mprop"
-  register: mprop_make
-
-- name: Make fat utility
-  shell: "make fat"
-  args:
-    chdir: "{{ siesta_topdir }}/Util/COOP"
-    creates: "{{ siesta_topdir }}/Util/COOP/fat"
-  register: fat_make
+# Install and configure libblacs-mpi-dev only for 16.04
+- name: Install and configure libblacs-mpi-dev for older distros
+  block:
 
-- name: Make gnubands utility
-  shell: "make gnubands"
-  args:
-    chdir: "{{ siesta_topdir }}/Util/Bands"
-    creates: "{{ siesta_topdir }}/Util/Bands/gnubands"
-  register: gnubands_make
+  - name: Install libblacs-mpi-dev
+    become: true
+    apt:
+      name:
+      - libblacs-mpi-dev
 
-- name: Make eigfat2plot utility
-  shell: "make eigfat2plot"
-  args:
-    chdir: "{{ siesta_topdir }}/Util/Bands"
-    creates: "{{ siesta_topdir }}/Util/Bands/eigfat2plot"
-  register: eigfat2plot_make
+  - name: Set blacs link string
+    set_fact:
+      siesta_blacs_libs: "-lblacsF77init-openmpi -lblacsCinit-openmpi -lblacs-openmpi"
 
-- import_tasks: tests.yml
-  when: siesta_make.changed and run_tests is defined and run_tests
+  when: ansible_facts['distribution_major_version'] | int <= 16
 
-- name: Install siesta executable
-  become: true
-  copy:
-    src: "{{ siesta_topdir }}/Obj/siesta"
-    dest: "{{ siesta_prefix }}/bin"
-    remote_src: true
-    mode: '0755'
-  register: siesta_install
-
-- name: Install mprop executable
-  become: true
-  copy:
-    src: "{{ siesta_topdir }}/Util/COOP/mprop"
-    dest: "{{ siesta_prefix }}/bin"
-    remote_src: true
-    mode: '0755'
-  register: mprop_install
+- name: Unset blacs link string for modern distros
+  set_fact:
+    siesta_blacs_libs: ""
+  when: ansible_facts['distribution_major_version'] | int > 16
 
-- name: Install fat executable
-  become: true
-  copy:
-    src: "{{ siesta_topdir }}/Util/COOP/fat"
-    dest: "{{ siesta_prefix }}/bin"
-    remote_src: true
-    mode: '0755'
-  register: fat_install
+# Compile and install flook
+- include: flook.yml
+  when: siesta_enable_flook | bool
 
-- name: Install gnubands executable
-  become: true
-  copy:
-    src: "{{ siesta_topdir }}/Util/Bands/gnubands"
-    dest: "{{ siesta_prefix }}/bin"
-    remote_src: true
-    mode: '0755'
-  register: gnubands_install
-
-- name: Install eigfat2plot executable
-  become: true
-  copy:
-    src: "{{ siesta_topdir }}/Util/Bands/eigfat2plot"
-    dest: "{{ siesta_prefix }}/bin"
-    remote_src: true
-    mode: '0755'
-  register: eigfat2plot_install
+# Compile and install xmlf90
+- include: xmlf90.yml
```

include is being deprecated: https://docs.ansible.com/ansible/2.9/modules/include_module.html

```suggestion
- include-tasks: xmlf90.yml
```

---

and the same for the others below